### ISSUE TITLE
spring.kafka.bootstrap-servers overrides brokers configuration of every binder.

### ISSUE THREAD

**Issue by DmitryRyzhikov** [spring-cloud/spring-cloud-stream-binder-kafka#967]
Hi,
I have to configure two separate kafka brokers (as on example below) and this is working just fine.

```
spring:
  cloud:
    stream:
      binders:
        kafka1:
          type: kafka
          environment:
            spring:
              cloud:
                stream:
                  kafka:
                    binder:
                      brokers: dev1-stage.dub:9092
        kafka2:
          type: kafka
          environment:
            spring:
              cloud:
                stream:
                  kafka:
                    binder:
                      brokers: dev2-stage.dub:9092
      bindings:
        firstOut:
          destination: first_topic
          binder: kafka1
        secondOut:
          destination: second_topic
          binder: kafka2
```
But also I need to configure zipkin+sleuth in application - and only way I can do for this for kafka sender type is via adding.
```
spring:
  kafka:
    bootstrap-servers: dev1-stage.dub:9092
```
As soon, as spring.kafka.bootstrap-servers added, it started to override kafka2.environment.spring.cloud.stream.kafka.binder.brokers - so application just trying to write to kafka on dev1-stage.dub, instead of dev2-stage.dub.

How can I prevent this overriding?

**Comment by sobychacko** [comment]
@DmitryRyzhikov It looks like when you provide `spring.kafka.bootstrap...` property, that overrides the binder specific property. We need to look into it further to see whats going on. We will get back.

**Comment by sobychacko** [comment]
@DmitryRyzhikov I am unable to reproduce the behavior that you are seeing. Could you try [this](https://github.com/spring-cloud/spring-cloud-stream-samples/tree/master/multi-binder-samples/multi-binder-two-kafka-clusters) sample? 

I added the following property to the `application.yml` of that application so that it simulates the same configuration that you have provided, i.e. Kafka1 broker address is duplicated for `spring.kafka.bootstrap-servers`. 

```
spring.kafka.bootstrap-servers: ${kafkaBroker1}
````

Then re-ran the sample and everything still works as expected. How is this different from your setup other than having different broker addresses? If you can reproduce the same behavior that you are seeing using this sample, that will make things easier to debug it further. Thanks!

**Comment by DmitryRyzhikov** [comment]
After hours of debug, looks that I was able to locate source of problem. 
Lib: spring-cloud-stream-binder-kafka:3.0.3.RELEASE.

KafkaMessageChannelBinder#createProducerMessageHandler:336.

In this method **KafkaTemplate** is created, that further is used on message sending phase. **KafkaTemplate.producerFactory.configs** is a map, that contains parameter under key **bootstrap.servers**. 

**DefaultKafkaProducerFactory** is created under org.springframework.cloud.stream.binder.kafka.KafkaMessageChannelBinder#getProducerFactory. And in this method we can find this call to **KafkaBinderConfigurationProperties**:

`
KafkaBinderConfigurationProperties#mergedProducerConfiguration:287

producerConfiguration.putAll(this.kafkaProperties.buildProducerProperties());
`
Line of code above adds all producer properties created on base of **KafkaProperties**. And this is first part of problem. If property **spring.kafka.bootstrap-servers** is provided in config, and it is not **localhost:9092**, it value will be returned here. But if property **spring.kafka.bootstrap-servers** is NOT provided in config, **localhost:9092** will be returned as default value here.

And after this we go to this piece of code - second part of the problem.
`
KafkaBinderConfigurationProperties.getConfigurationWithBootstrapServer:

				if (bootStrapServers.size() == 1
						&& bootStrapServers.get(0).equals("localhost:9092")) {
					configuration.put(bootstrapServersConfig, getKafkaConnectionString());
				}
`
Basically, if property **spring.kafka.bootstrap-servers** is provided in config, then code inside if block is skipped - and value of this property IS NOT OVERRIDEN by initial binder configuration and goes to map **configs** under key **bootstrap.servers**. If **spring.kafka.bootstrap-servers** property is not provided, then default value "localhost:9092" here will be overriden by original binder **brokers** property - just as I expect.

For me this looks as not expected behavior, because, as I wrote in initial post, I have to provide **spring.kafka.bootstrap-servers** - Zipkin expects this property.
